tesseract head with opencl ""_clReleaseProgram", referenced from:"

```
$ brew install tesseract --HEAD --with-opencl
==> Using the sandbox
==> Cloning https://github.com/tesseract-ocr/tesseract.git
Updating /Users/latistaadmin/Library/Caches/Homebrew/tesseract--git
==> Checking out branch master
==> ./autogen.sh
==> ./configure --prefix=/usr/local/Cellar/tesseract/HEAD-a75ab45_2 --enable-opencl
==> make install
Last 15 lines from /Users/latistaadmin/Library/Logs/Homebrew/tesseract/03.make:
  "_clReleaseProgram", referenced from:
      OpenclDevice::ReleaseOpenclEnv(_GPUEnv*) in libtesseract_opencl.a(openclwrapper.o)
  "_clSetKernelArg", referenced from:
      OpenclDevice::pixReadFromTiffKernel(unsigned int*, int, int, int, unsigned int*) in libtesseract_opencl.a(openclwrapper.o)
      pixDilateCL_55(int, int) in libtesseract_opencl.a(openclwrapper.o)
      pixErodeCL_55(int, int) in libtesseract_opencl.a(openclwrapper.o)
      pixDilateCL(int, int, int, int) in libtesseract_opencl.a(openclwrapper.o)
      pixErodeCL(int, int, unsigned int, unsigned int) in libtesseract_opencl.a(openclwrapper.o)
      pixORCL_work(unsigned int, unsigned int, _cl_mem*, _cl_mem*, _cl_mem*) in libtesseract_opencl.a(openclwrapper.o)
      pixANDCL_work(unsigned int, unsigned int, _cl_mem*, _cl_mem*, _cl_mem*) in libtesseract_opencl.a(openclwrapper.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [libtesseract.la] Error 1
make: *** [install-recursive] Error 1
```

**brew gist-logs tesseract**
https://gist.github.com/y-fedorov/0e5a302dd68ff9d8647f03ba4c15050a

**brew update**
Already up-to-date.

**$ cd "$(brew --repo)" && git fetch && git reset --hard origin/master && brew update**
HEAD is now at 5a264cf Merge pull request #947 from zmwangx/fix-test-bot-desc
Updated 1 tap (homebrew/core).
==> Updated Formulae
redpen

$ brew install tesseract   - is OK
$ brew install tesseract --HEAD   - is OK

Yup,
```
==> Installing tesseract
==> Using the sandbox
==> Cloning https://github.com/tesseract-ocr/tesseract.git
Cloning into '/Users/joe/Library/Caches/Homebrew/tesseract--git'...
remote: Counting objects: 794, done.
remote: Compressing objects: 100% (733/733), done.
remote: Total 794 (delta 125), reused 235 (delta 42), pack-reused 0
Receiving objects: 100% (794/794), 3.78 MiB | 1.10 MiB/s, done.
Resolving deltas: 100% (125/125), done.
Checking connectivity... done.
==> Checking out branch master
==> ./autogen.sh
==> ./configure --prefix=/usr/local/Cellar/tesseract/HEAD-a75ab45_2 --enable-opencl
==> make install
Last 15 lines from /Users/joe/Library/Logs/Homebrew/tesseract/03.make:
  "_clReleaseProgram", referenced from:
      OpenclDevice::ReleaseOpenclEnv(_GPUEnv*) in libtesseract_opencl.a(openclwrapper.o)
  "_clSetKernelArg", referenced from:
      OpenclDevice::pixReadFromTiffKernel(unsigned int*, int, int, int, unsigned int*) in libtesseract_opencl.a(openclwrapper.o)
      pixDilateCL_55(int, int) in libtesseract_opencl.a(openclwrapper.o)
      pixErodeCL_55(int, int) in libtesseract_opencl.a(openclwrapper.o)
      pixDilateCL(int, int, int, int) in libtesseract_opencl.a(openclwrapper.o)
      pixErodeCL(int, int, unsigned int, unsigned int) in libtesseract_opencl.a(openclwrapper.o)
      pixORCL_work(unsigned int, unsigned int, _cl_mem*, _cl_mem*, _cl_mem*) in libtesseract_opencl.a(openclwrapper.o)
      pixANDCL_work(unsigned int, unsigned int, _cl_mem*, _cl_mem*, _cl_mem*) in libtesseract_opencl.a(openclwrapper.o)
      ...
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [libtesseract.la] Error 1
make: *** [install-recursive] Error 1

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/brew):
  https://github.com/Homebrew/homebrew-core/issues

These open issues may also help:
tesseract head with opencl ""_clReleaseProgram", referenced from:" https://github.com/Homebrew/homebrew-core/issues/4764
```

--- a/Formula/tesseract.rb
+++ b/Formula/tesseract.rb
@@ -72,6 +72,14 @@ class Tesseract < Formula
   end
 
   def install
+    if build.head?
+      # ld: symbol(s) not found for _clSetKernelArg and other symbols
+      # Regression caused by https://github.com/tesseract-ocr/tesseract/commit/b1c921b
+      # Reported 13 Sep 2016 https://github.com/tesseract-ocr/tesseract/issues/426
+      inreplace "api/Makefile.am", "$(GENERIC_LIBRARY_VERSION) -no-undefined",
+                                   "$(GENERIC_LIBRARY_VERSION)"
+    end
+
     if build.with? "training-tools"
       icu4c = Formula["icu4c"]
       ENV.append "CFLAGS", "-I#{icu4c.opt_include}"

